src/vs/editor/contrib/codeAction/codeActionModel.ts

```diff
@@ -19,6 +19,7 @@ import { CodeActionKind, CodeActionTrigger } from './codeActionTrigger';
 
 export const HAS_REFACTOR_PROVIDER = new RawContextKey<boolean>('hasRefactorProvider', false);
 export const HAS_SOURCE_ACTION_PROVIDER = new RawContextKey<boolean>('hasSourceActionProvider', false);
+export const HAS_ORGANIZE_IMPORTS_PROVIDER = new RawContextKey<boolean>('hasOrganizeImportsActionProvider', false);
```

Comment:
Can we have a more generic context key? We could use equality or the regex support. Something like `supportedCodeAction ~= /source\.organizeImports.*/`

Reply:
Done. `supportedCodeAction` is a space separated list of provided kinds. The regexp you use will typically look like:

```js
/\bsource\.organizeImports(\b|\.)/
```

Which will match `source.organizeImports` as well as `source.organizeImports.alphabetically`